### DISCUSSION TITLE
Issues with pyinstaller

### DISCUSSION THREAD

**Discussion by thomasrunge** [mdomke/schwifty#184] [Q&A]
Schwifty doesn't work nicely with pyinstaller by default when using the "--onefile"-option. Here is a solution that works for me, just in case someone is interested.

In your pyinstaller-spec file add this:
```
from PyInstaller.utils.hooks import copy_metadata, collect_data_files
```
and use
```
datas = copy_metadata("schwifty") + collect_data_files("schwifty")
````

This copies the generated json files ("bank_registry" and "iban_registry") to your executable. As this adds a subdirectory with the name of the package ("schwifty") and our own directories below that, we need to adjust registry.py: change line 53
from
```
directory = package_path / f"{name}_registry"
```
to
```
directory = package_path / "schwifty" / f"{name}_registry"
```

**Comment by thomasrunge** [comment]
This fixes runs with pyinstaller-made exe files, but disturbs non-pyinstaller-runs. The better (correct?) solution now works fine for me in every scenario:

In schwifty\registry.py line 48:
```
package_path = Path(getattr(sys, "_MEIPASS"))
```
change to:
```
package_path = Path(getattr(sys, "_MEIPASS")) / __package__
```

**Comment by mdomke** [comment]
The issue should be fixed in `schwifty` release `2024.05.2` where #92 has been reverted as suggested by the `pyinstaller` maintainer here https://github.com/mdomke/schwifty/issues/196#issuecomment-2102330388
